libraries/SITL/SIM_GPS.h

```diff
@@ -69,28 +69,24 @@ class GPS_Backend {
     GPS_Backend(class GPS &front, uint8_t _instance);
     virtual ~GPS_Backend() {}
 
-    void update(const GPS_Data &d);
-
     // 0 baud means "unset" i.e. baud-rate checks should not apply
     virtual uint32_t device_baud() const { return 0; }
 
     ssize_t write_to_autopilot(const char *p, size_t size) const;
     ssize_t read_from_autopilot(char *buffer, size_t size) const;
 
+    // read and process config from autopilot (e.g.)
+    virtual void update_read();
+    // writing fix information to autopilot (e.g.)
+    virtual void update_write(const GPS_Data *d) = 0;
```

Comment:
could we pass this by reference instead of by pointer?  I think we normally try to pass by reference if possible perhaps because this slightly reduces the chance of passing a nullptr.

Reply:
Yea, I can investigate that next time if that's possible. Great suggestion.